pkg/destroy/providers/registry.go

```diff
@@ -0,0 +1,19 @@
+package providers
+
+import (
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+// Destroyer allows multiple implementations of destroy
+// for different platforms.
+type Destroyer interface {
+	Run() error
+}
+
+// NewFunc is an interface for creating platform-specific destroyers.
+type NewFunc func(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (Destroyer, error)
+
+// Registry maps ClusterMetadata.Platform() to per-platform Destroyer creators.
+var Registry = make(map[string]NewFunc)
```

Comment:
nit: except this variable. other definition don't belong to this file.

Reply:
moved to different file